extensions/gitpod/src/extension.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import Log from './common/logger';
+import GitpodAuthenticationProvider from './authentication';
+import LocalApp from './localApp';
+import { enableSettingsSync, updateSyncContext } from './settingsSync';
+import { GitpodServer } from './gitpodServer';
+import TelemetryReporter from './telemetryReporter';
+import { collectLogs } from './collectLogs';
+
+const EXTENSION_ID = 'gitpod.gitpod-desktop';
+const ANALITYCS_KEY = 'bUY8IRdJ42KjLOBS9LoIHMYFBD8rSzjU';
+
+let telemetry: TelemetryReporter;
+
+export async function activate(context: vscode.ExtensionContext) {
+	const logger = new Log('Gitpod');
+
+	const version = vscode.extensions.getExtension(EXTENSION_ID)!.packageJSON.version;
+	telemetry = new TelemetryReporter(EXTENSION_ID, version, ANALITYCS_KEY);
+
+	/* Gitpod settings sync */
+	await updateSyncContext();
+	context.subscriptions.push(vscode.workspace.onDidChangeConfiguration(e => {
+		if (e.affectsConfiguration('gitpod.host') || e.affectsConfiguration('configurationSync.store')) {
+			updateSyncContext();
+		}
+	}));
+
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.syncProvider.remove', async () => {
+		try {
+			await enableSettingsSync(false, telemetry);
+		} catch (e) {
+			const outputMessage = `Error setting up Settings Sync with Gitpod: ${e}`;
+			vscode.window.showErrorMessage(outputMessage);
+			logger.error(outputMessage);
+		}
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.syncProvider.add', async () => {
+		try {
+			await enableSettingsSync(true, telemetry);
+		} catch (e) {
+			const outputMessage = `Error setting up Settings Sync with Gitpod: ${e}`;
+			vscode.window.showErrorMessage(outputMessage);
+			logger.error(outputMessage);
+		}
+	}));
+
+	await vscode.commands.executeCommand('setContext', 'gitpod.remoteName', vscode.env.remoteName);
```

Comment:
I don't think it is necessary, this command should work even without remote and then collect only local logs, i.e. to troubleshoot for instance local setttings sync.